libs/ledger/ledger-utxo-transaction-verifier/src/test/kotlin/net/corda/ledger/utxo/transaction/verifier/UtxoLedgerTransactionVerifierTest.kt

```diff
@@ -132,32 +134,38 @@ class UtxoLedgerTransactionVerifierTest {
     }
 
     @Test
-    fun `throws an exception if the notary is not allowed`() {
-        // TODO CORE-8956 Check the notary is in the group parameters whitelist
-    }
-
-    @Test
-    fun `throws an exception when input and reference states don't have the same notary`() {
+    fun `throws an exception when input and reference states don't have the same notary (names are different)`() {
         whenever(inputTransactionState.notaryName).thenReturn(notaryX500Name)
-        whenever(inputTransactionState.notaryKey).thenReturn(publicKeyExample)
         whenever(referenceTransactionState.notaryName).thenReturn(anotherNotaryX500Name)
-        whenever(referenceTransactionState.notaryKey).thenReturn(anotherPublicKeyExample)
         assertThatThrownBy { verifier.verify() }
             .isExactlyInstanceOf(IllegalStateException::class.java)
             .hasMessageContaining("Input and reference states' notaries need to be the same.")
     }
 
     @Test
-    fun `throws an exception when input and reference states don't have the same notary passed into the verification`() {
+    fun `does not throw when input and reference states don't have the same notary keys (but the names are still the same)`() {
+        whenever(inputTransactionState.notaryKey).thenReturn(publicKeyExample)
+        whenever(referenceTransactionState.notaryKey).thenReturn(anotherPublicKeyExample)
+        assertDoesNotThrow { verifier.verify() }
+    }
+
+
+    @Test
+    fun `throws an exception when input and reference states don't have the same notary passed into the verification (names are different)`() {
         whenever(inputTransactionState.notaryName).thenReturn(anotherNotaryX500Name)
-        whenever(inputTransactionState.notaryKey).thenReturn(anotherPublicKeyExample)
         whenever(referenceTransactionState.notaryName).thenReturn(anotherNotaryX500Name)
-        whenever(referenceTransactionState.notaryKey).thenReturn(anotherPublicKeyExample)
         assertThatThrownBy { verifier.verify() }
             .isExactlyInstanceOf(IllegalStateException::class.java)
             .hasMessageContaining("Input and reference states' notaries need to be the same as the UtxoLedgerTransaction's notary")
     }
 
+    @Test
+    fun `does not throw when input and reference states don't have the same notary keys passed into the verification (but the names are still the same)`() {
```

Comment:
```suggestion
    fun `does not throw when input and reference states have different notary keys passed into verification (names are the same)`() {
```

Means you don't need the long line limit suppression.

Reply:
Thanks. Updated.